pennylane/templates/subroutines/permute.py

```diff
@@ -157,15 +161,62 @@ def __init__(self, permutation, wires, do_queue=True, id=None):
             # Make sure everything in the permutation has an associated label in wires
         for label in permutation:
             if label not in wires:
-                raise ValueError(f"Cannot permute wire {label} not present in wire set.")
+                raise ValueError(
+                    f"Cannot permute wire {label} not present in wire set.")
 
-        self._hyperparameters = {"permutation": permutation}
+        self._hyperparameters = {"permutation": permutation, 'wires': wires}
         super().__init__(wires=wires, do_queue=do_queue, id=id)
 
     @property
     def num_params(self):
         return 0
 
+    def matrix(self):  # pylint: disable=arguments-differ
+
+        wires = self._hyperparameters['wires']
+        permutation = self._hyperparameters['permutation']
```

Comment:
We can use the public `hyperparameters` property for more clarity.
```suggestion
        permutation = self.hyperparameters['permutation']
```